src/diff_tform.c

```diff
@@ -185,15 +146,16 @@ int git_diff_merge(git_diff *onto, const git_diff *from)
 			STRCMP_CASESELECT(ignore_case, o->old_file.path, f->old_file.path);
 
 		if (cmp < 0) {
-			delta = diff_delta__dup(o, &onto_pool);
+			delta = git_diff__delta_dup(o, &onto_pool);
 			i++;
 		} else if (cmp > 0) {
-			delta = diff_delta__dup(f, &onto_pool);
+			delta = git_diff__delta_dup(f, &onto_pool);
 			j++;
 		} else {
-			delta = reversed ?
-				diff_delta__merge_like_cgit_reversed(o, f, &onto_pool) :
-				diff_delta__merge_like_cgit(o, f, &onto_pool);
+			const git_diff_delta *left = reversed ? f : o;
+			const git_diff_delta *right = reversed ? o : f;
+
+			delta = cb(left, right, &onto_pool);
```

Comment:
So when I working on this code, I wondered myself why `diff_delta__merge_like_cgit_reversed()` even existed vs just swapping the parameters to `diff_delta__merge_like_cgit()`. If you look at the implementation of `diff_delta__merge_like_cgit_reversed()`, it's actually not exactly the "reverse" of `diff_delta__merge_like_cgit` IIRC. Are you sure this is safe?